custom_components/myskoda/coordinator.py

```diff
@@ -168,38 +169,35 @@ async def _on_operation_event(self, event: EventOperation) -> None:
             await self.update_status(immediate=True)
 
     async def _on_charging_event(self, event: EventCharging):
-        # TODO @webspider: Refactor with proper classes
         vehicle = self.data.vehicle
-
         data = event.event.data
 
         if vehicle.charging is None or vehicle.charging.status is None:
             await self.update_charging()
         else:
```

Comment:
offtopic: I actually don't think this should be an `else` structure. If we don't have charging data, we should populate it and continue. Maybe that fits better in another PR.

Reply:
Yes, to be handled in #370